scala/StalinSort.scala

```diff
@@ -0,0 +1,15 @@
+object StalinSort extends App {
+
+  def stalinSort[A: Ordering](xs: List[A]): List[A] = xs match {
+    case Nil          => Nil
+    case x :: Nil     => List(x)
+    case x :: y :: zs =>
+      if (implicitly[Ordering[A]].lteq(x, y)) x :: stalinSort(y :: zs)
```

Comment:
Beat me to the Scala version!! 😛 

For the comparison, why not:

```scala
case x :: y :: zs => if (x < y) then x :: stalinSort(y :: zs) else stalinSort(x :: zs)
```

Reply:
You are right for a concrete implementation but, this one is polymorphic.

```

scala>   def stalinSort[A: Ordering](xs: List[A]): List[A] = xs match {
     |     case Nil => Nil
     |     case x :: Nil => List(x)
     |     case x :: y :: zs =>
     |       if(x < y) x :: stalinSort(y :: zs) else stalinSort(x :: zs)
     | }
<console>:15: error: value < is not a member of type parameter A
             if(x < y) x :: stalinSort(y :: zs) else stalinSort(x :: zs)
                  ^
```

Reply:
Thanks 👍